src/controllers/ajax/IndexAjaxController.php

```diff
@@ -119,6 +131,25 @@ protected function getActions(): array<string> {
 
     // Check logo
     $final_logo = $logo;
+
+    if ($is_custom_logo) {
+      $logo_data = str_replace(' ', '+', $final_logo);
+      $logo_data_unbase64ed = base64_decode($logo_data);
+
+      $name = 'custom-'.time();
```

Comment:
A bit more of entropy here would be nice, just the timestamp won't do it. maybe md5 the filename and get a chunk of it?